jenkinsbuilder/cccp_index_reader.py

```diff
@@ -110,15 +112,25 @@ def main(indexdlocation):
                                   ':'.join(
                                       [jjb_defaults_file, generated_filename])
                                   ]
+                        if mock:
+                            ret_data.append(
+                                {
+                                    "project_info": new_proj,
+                                    "dump_file": generated_filename,
+                                    "cmd": myargs
+                                }
+                            )
                         print myargs
-                        proc = subprocess.Popen(myargs,
-                                                stdout=subprocess.PIPE)
-                        proc.communicate()
+                        if not mock:
+                            proc = subprocess.Popen(myargs,
+                                                    stdout=subprocess.PIPE)
+                            proc.communicate()
 
                     finally:
                         print "Removing {}".format(t)
                         # shutil.rmtree(t)
-
+                        if mock:
+                            return ret_data
```

Comment:
:+1: for adding ``if mock``. Consider using stubs, if needed.

I think that the function should spit the same data as output when called from test or from application code. It depends on the callee to decide what to do with the return data. It'd be great, not to add 'mocking' logic into the code being tested. This will add unnecessary complexity. It'd be great to break a big function into smaller ones which takes some data structure (dict or list) input and returns some output on which you can do some assertion on it.